tests.js

```diff
@@ -1,44 +1,103 @@
 require('dotenv').load();
+
+var _ = require('underscore');
+var chai = require('chai');
+var fs = require('fs');
 var FreshBooks = require('./freshbooks.js');
-var assert = require('chai').assert;
+var request = require('request');
 
-describe('FreshBooks Client', function() {
-  var freshbooks;
-  before(function() {
-    var access_token = process.env.FBOOKS_ACCESS_TOKEN;
-    var refresh_token = process.env.FBOOKS_REFRESH_TOKEN;
-    if (!access_token || !refresh_token)
-      throw new Error('Required environment variables FBOOKS_ACCESS_TOKEN and/or FBOOKS_REFRESH_TOKEN were not populated');
+describe('freshbooks', function() {
+  this.timeout(1000 * 20);
 
-    freshbooks = new FreshBooks(access_token, refresh_token);
+  var token;
+  var freshbooks;
+  var biz_id, project_id, time_entry_id;
+  it('should get an access token correctly', function(done) {
+    var code = fs.readFileSync('code.txt').toString();
+    request({
+      method: 'POST',
+      url: process.env.token_url,
+      headers: {
+        'content-type': 'application/json',
+        'Api-Version': 'alpha'
+      },
+      json: {
+        grant_type: 'authorization_code',
+        client_secret: process.env.client_secret,
+        code: code,
+        client_id: process.env.client_id,
+        redirect_uri: 'https://localhost:8081/fbooks-callback'
+      }
+    }, function(err, response, body) {
+      if (!body.refresh_token) {
+        done(new Error(body.error + ' ' + JSON.stringify(body)));
+      }
+      else {
+        freshbooks = new FreshBooks(body.access_token, body.refresh_token);
+        done();
+      }
+    });
   });
 
   it('should get the current user data', function(done) {
     freshbooks.me(function(err, me) {
       if (err) return done(err);
-      console.log(JSON.stringify(me));
+
+      chai.assert.ok(me.id);
       done();
     });
   });
 
   it('should get the projects for a business', function(done) {
     freshbooks.me(function(err, me) {
       if (err) return done(err);
-      var biz_id = me.business_ids[0];
+      biz_id = me.business_ids[0];
       freshbooks.getProjects(biz_id, function(err, projects) {
         if (err) return done(err);
+
+        project_id = projects[0].id;
+        chai.assert.ok(projects.length);
+        chai.assert.ok(projects[0].id);
```

Comment:
@schuttsm: This is a great start, but I think we should verify the presence & the type of every property we're using (or we should stop using it if we don't need to):

In `bc-proxy.js`, `loadProjectsForBiz()` I think that also includes:

* `body.meta.pages` and `body.meta.page` (i think we're expecting both to be numbers)
* `proj.client_id`
* `proj.title`
* `proj.updated_at`
* `proj.active` (looks like we're expecting a boolean?)

Also, `getActiveBusinesses()` in `api-utils.js` depends on some `subscription_statuses` and `business_memberships` data (I think originally we didn't have this, but got errors from the API when we attempted to pull data from businesses whose subscription has lapsed). This data is being saved to the Firebase `people/{psn_id}/service_data/freshbooks` by the `/fbooks-callback` route in `bcsyncer/index.js`. Also, we have *got* to consolidate as much of the FreshBooks API code into this lib as possible, I'm going cross-eyed trying to find it across all the different containers it's scattered in. I'm sorry about the mess, most of it made some kind of sense at the time (though I admit some of it was pragmatic).